include/albatross/src/core/model.hpp

```diff
@@ -17,6 +17,8 @@ namespace albatross {
 
 using Insights = std::map<std::string, std::string>;
 
+constexpr bool DEFAULT_USE_ASYNC = false;
```

Comment:
In previous conversation we decided not to put an `async` in model base, I think I'd still prefer to keep it isolated to the sparse GP (which is the only thing that uses it at the moment) and include it in all models once we get a critical mass using it.  Was there an issue that came up that changed that decision?

Reply:
It makes it much more difficult to add upstream configurability. We perform model_interface lookup by name, without configuration options. Changing configuration afterwards requires compatibility with the NullModel, which is really only possible if more than the SparseGP changed.
An alternative is adding toggling async by adding it to the model name string. A second alternative is passing a config to interface creation, but that will require decloning functionality shared between our 3 main model types in OE.
Really, my choice here was all about the OE interaction.

Reply:
I think I remember, doing it for only the sparse GP makes it so you subsequently need to special case `NullModel` etc if you want to set the async flag for a generic model.  I'm fine with this as is for now and we can consider alternatives down the line.